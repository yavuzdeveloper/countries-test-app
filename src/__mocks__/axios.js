// export default {
//     get: jest.fn().mockResolvedValue({ 
//     countries : [] })
// };

export default {
  get: jest.fn().mockResolvedValue({ country: {} })
};