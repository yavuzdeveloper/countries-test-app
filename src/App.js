import React, { useEffect, useState } from "react";
import "./styles.css";
import axios from "axios";
//import Countries from "./Countries"
import Country from "./Country"


export default function App() {
  const [countries, setCountries] = useState([]);
  const [loaded, setLoaded] = useState(false);

  
  const fetchCountries = async () => {
    await axios
      .get("https://restcountries.eu/rest/v2/all")
      .then(response => {
        setCountries(response.data)
         setLoaded(true)
        })
      .catch(error => { 
        console.log({ error });
        setLoaded(false);
      });
  }

  useEffect(() => {
    // axios
    //   .get("https://restcountries.eu/rest/v2/all")
    //   .then(response => 
    //     setCountries(response.data))
    //   .catch(error => console.log({ error }));

  fetchCountries();

  }, []);



  return (
    <div className="App">
      <h1> 
        <img 
          src="https://avatars3.githubusercontent.com/u/60869810?v=4" 
          alt="react-logo" 
          style={{ width: "20px", padding:"0 10px 0 0" }}
        />
        Countries
      </h1>
 {/* <Countries />  */}


      { loaded ? ( <div> 
      { countries &&  countries.map(country =>
        <Country  country={country}/> 
      )}
    </div>
    ) : ( <p>
          <img 
            src= "https://media.giphy.com/media/y1ZBcOGOOtlpC/source.gif" 
            alt="imgLoading"       
          />
        </p>)
      }
    </div>
  );
}


//   <span data-testid="resolved" >{"countries"}</span>

//     { loaded ? ( <div> 
// { countries.map(country =>
//         <Country  country={country}/> 
// )}
//     </div>
     
//     ) : ( <p data-testid="loading"> Loading...
//           {/* <img 
//             src= "https://media.giphy.com/media/y1ZBcOGOOtlpC/source.gif" 
//             alt="imgLoading"       
//           /> */}
//         </p>)
//     }