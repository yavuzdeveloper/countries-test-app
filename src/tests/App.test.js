import React from 'react';
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

import axios from 'axios';
import TestRenderer from "react-test-renderer";

import App from "../App"
import { describe } from 'yargs';

//1 test:
test('1-App renders correctly', () => {
   render(<App />);
});

//2 test:
// describe("2-React logo", () => {
//     test("2-React logo", () => {
//         render(<App />);
//             const logoReact = screen.getByRole("img");

//             expect(logoReact).toHaveAttribute("src", 'https://avatars3.githubusercontent.com/u/60869810?v=4')
//             expect(logoReact).toHaveAttribute('alt', "react-logo");
//      });
// });

//3.test:
// describe("Img loading", () => {
//     test("3-Img loading", () => {
//        render(<App />);
//         const logoReact = screen.getByRole("img");

//         expect(logoReact).toHaveAttribute("src", "https://media.giphy.com/media/y1ZBcOGOOtlpC/source.gif")
//         expect(logoReact).toHaveAttribute('alt', "imgLoading" );
//     });
// });




//3 test:
test("4-title renders correctly", () => {
    render(<App />);
       const titleElement = screen.getByText(/countries/i);
    //    expect(titleElement).toBeInTheDocument(); //doc olmasını bekliyoruz.
    //    expect(titleElement).toHaveTextContent(/countries/i);
});




// async function getUsers() {
//     try {
//         const response = await axiosMock.get("/users");
//         return response.data.users;
//     } catch (e) {
//         throw new Error("Oops. Something wrong happened");
//     }
// }

// jest.mock("axios");

// const fakeUsers = ["john", "Emma", "Tom"];
// axiosMock.get.mockResolvedValue({ data: { users: fakeUsers } });

// test("get the users", async () => {
//     const users = await getUsers();
//     expect(users).toEqual(fakeUsers);
// });


//axios-mock;

// afterEach(cleanup);

// jest.mock('axios');// added??

// test('mocking axios request', async () => {
//   const data = {
//     countries: [
//       { name: 'country1' },
//       { name: 'country2' },
//       { name: 'country3' }
//     ]
//   };

//   mockedAxios.get.mockResolvedValueOnce(data);
//   const { getByText } = render(<App />);

//   await waitFor(() => {
//     expect(getByText('country1')).toBeInTheDocument();
//   });
// });

//***************
jest.mock('axios');

test("fetch", () => {
  const response = {
    countries: [
      { name: 'country1' },
      { name: 'country2' },
      { name: 'country3' }
    ]
  }
  axios.get.mockResolvedValue(response);

    const component = TestRenderer.create(
      <App />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});