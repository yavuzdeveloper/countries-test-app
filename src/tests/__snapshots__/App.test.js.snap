// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch 1`] = `
<div
  className="App"
>
  <h1>
    <img
      alt="react-logo"
      src="https://avatars3.githubusercontent.com/u/60869810?v=4"
      style={
        Object {
          "padding": "0 10px 0 0",
          "width": "20px",
        }
      }
    />
    Countries
  </h1>
  <p>
    <img
      alt="imgLoading"
      src="https://media.giphy.com/media/y1ZBcOGOOtlpC/source.gif"
    />
  </p>
</div>
`;
